enhancements/audit-command.md

```diff
@@ -0,0 +1,722 @@
+  -v, --validation bool          set false to disable the bundle validation checks. (default true). 
+  -d, --data                     set false to disable the data report. It is useful for who would like to audit the index via only run the bundle and scorecard checks. (default true).
+  -h, --head-only                set the flag to audit only the operator bundles which are the head of the channel. (default all bundles of the index catalog image will be used)
+  -w, --warning                  set false to disable the report of warning(s) found in the checks (default warning and errors will be reported)
+```
+
+Note that the command will:
+- Use the index image to gathering the data
+- Download the operator bundles
+- Use SDK commands to run the checks 
+- Output the results
+
+**NOTE** If we move forward with this command as part of opm it would be like: `opm alpha index audit --help`
+ 
+### User Stories 
+ 
+- As an admin and or maintainer of OLM index catalog, I would like to check if all bundle operators in the catalogue are respecting all expected criteria so that, I can ensure a better quality of all bundles in the catalogue.
+- As an admin and or maintainer of OLM index catalog, I would like to easily know more about the bundles in the catalogue so that, I can easily provide strategies which will be useful for the biggest part of the authors of the projects published on it.
+- As an admin and or maintainer of OLM index catalog, identify problems and concerns in the operators bundles of some catalogue so that, I can easily decide if they should or not be removed from the index catalog.
+- As an admin and or maintainer of OLM index catalog I would like to audit the OLM index catalog by running custom tests for each operator bundle that, I can use it to verify if all operator bundles will work as expected.
+- **(Optional)** As operator developer I would like to configure scorecard via SDK in a bundle of a project which is a NO-SDK layout so that, I could check my bundle with scorecard feature. For more info check [Scorecard config manifests](#scorecard-config-manifests). We can address it in the audit command as well at the first moment with some implications. Then, it is not required at all if the solution get shipped on SDK too.
+
+### Implementation Details/Notes/Constraints 
+
+Following the actions which are being proposed in this document for the new command:
+
+- Extract the database from the image informed
+- Via sql get all bundles name and paths from the index catalog image
+- Download and extract all bundles files by using the operator bundle path
+- Get the required data for the report via sql from the index, from the operator bundle manifest files and from the operator repository
+- Use sdk tool to execute the bundle validate checks
+- Use sdk tool to execute the scorecard bundle checks
```

Comment:
Seems like we should extract the necessary bits so that they can be reused as a library.

Reply:
We have a lib for that already. the implementation used by the feature is done in framework/api. However, would imo not make sense to re-write the sdk implementation for that instead of just use it via a bin dep ass for example.

Reply:
Clarified as well in the EP and in many similar comments here. So, I am closing this as sorted out.

Reply:
We have the validators and the scorecard config implemented in the operator-framework/api. then, we would need basically re-write the scorecard and bundle validate features (sdk code implementation) for the audit as well.